src/ai/index.js

```diff
@@ -0,0 +1,533 @@
+createNameSpace("realityEditor.ai");
+
+import * as THREE from '../../thirdPartyCode/three/three.module.js';
+
+(function(exports) {
+    
+    let question1 = '';
+    let categorize_prompt = 'Which one of the following items best describes my question? 1. "summary", 2. "debug", 3. "tools", 4. "pdf", 5. "tool content", 6. "not relevant". You can only return one of these items in string.';
+    
+    function setupSystemEventListeners() {
+        map.setupEventListeners();
+        realityEditor.gui.pocket.registerCallback('frameAdded', (params) => {
+            let avatarId = realityEditor.avatar.getAvatarObjectKeyFromSessionId(globalStates.tempUuid);
+            onFrameAdded(params, avatarId);
+        });
+        // todo Steve: add tool reposition event triggering for the user who added the tool themselves
+        realityEditor.device.registerCallback('vehicleDeleted', (params) => {
+            let avatarId = realityEditor.avatar.getAvatarObjectKeyFromSessionId(globalStates.tempUuid);
+            onVehicleDeleted(params, avatarId);
+        });
+    }
+    
+    function focusOnFrame(frameKey) {
+        let framePosition = realityEditor.gui.threejsScene.getToolPosition(frameKey);
+        let cameraPosition = realityEditor.gui.threejsScene.getCameraPosition();
+        let frameDirection = cameraPosition.clone().sub(framePosition).normalize();
+        realityEditor.device.desktopCamera.focusVirtualCamera(framePosition, frameDirection);
+    }
+    
+    function onFrameAdded(params, avatarId = 'Anonymous id') {
+        let objectId = params.objectKey;
+        let frameId = params.frameKey;
+        let frameType = params.frameType;
+        let frame = realityEditor.getFrame(objectId, frameId);
+        
+        let m = frame.ar.matrix;
+        let position = new THREE.Vector3(m[12], m[13], m[14]);
+        let groundPlaneMatrix = realityEditor.sceneGraph.getGroundPlaneNode().worldMatrix;
+        let inverseGroundPlaneMatrix = new realityEditor.gui.threejsScene.THREE.Matrix4();
+        realityEditor.gui.threejsScene.setMatrixFromArray(inverseGroundPlaneMatrix, groundPlaneMatrix);
+        inverseGroundPlaneMatrix.invert();
+        position.applyMatrix4(inverseGroundPlaneMatrix);
+
+        let timestamp = getFormattedTime();
+        let avatarName = realityEditor.avatar.getAvatarNameFromObjectKey(avatarId);
+        let avatarScrambledId = realityEditor.ai.crc.generateChecksum(avatarId);
+        let frameScrambledId = realityEditor.ai.crc.generateChecksum(frameId);
+        map.addToMap(avatarId, avatarName, avatarScrambledId);
+        map.addToMap(frameId, frameType, frameScrambledId);
+        let newInfo = `User ${avatarScrambledId} added a ${frameScrambledId} tool at ${timestamp} at (${position.x.toFixed(0)},${position.y.toFixed(0)},${position.z.toFixed(0)})`;
```

Comment:
Yes --- I do plan to leave this information in for future reference. But another way is to directly query object.json and get the coords from there. Probably a future route that I'm going to take, since the current setup doesn't have memory after refreshing the page. Probably need to add more info to object.json tho, eg: who edited the tool, etc.